lib/mix/tasks/thrift.generate.ex

```diff
@@ -0,0 +1,30 @@
+defmodule Mix.Tasks.Thrift.Generate do
+  use Mix.Task
+
+  @moduledoc """
+  Generate Elixir modules from Thrift schema definitions.
+
+  ## Command line options
+
+    * `--thrift-dir` - Directory to scan for .thrift files. (Default: ./thrift)
+    * `--lib-dir` - Directory under which to place generated .ex files. (Default: ./lib)
+  """
+
+  def run(args) do
+    {opts, _, _} = OptionParser.parse(args)
+
+    thrift_dir = Keyword.get(opts, :thrift_dir, "thrift")
+    lib_dir = Keyword.get(opts, :lib_dir, "lib")
+
+    for thrift_file <- Mix.Utils.extract_files([thrift_dir], "*.thrift") do
```

Comment:
You can just pass `"thrift"` as the argument. `extract_files` can take a list of file extensions or a glob pattern.